package.json

```diff
@@ -3,27 +3,28 @@
   "version": "0.1.0",
   "description": "AsyncAPI website",
   "private": true,
+  "type": "module",
   "scripts": {
     "dev": "npm run build-scripts && next dev",
     "build": "npm run build-scripts && next build",
     "test": "jest --passWithNoTests",
-    "build:pages": "node scripts/build-pages.js && npm run format:mdx",
-    "build:posts": "node scripts/index.js",
+    "build:pages": "node --loader ts-node/esm scripts/build-pages.ts && npm run format:mdx",
```

Comment:
We discussed to use tsx as well. Why you again moved to ts-node?

Reply:
@akshatnema We can execute all the scripts with node + ts-node/esm loader.

---
### Problems with ts-node

I was getting the error `Cannot use import outside a module`. To resolve this, I added the `type: "module"` configuration in the `package.json` file. However, this caused another error: `.ts is not a recognized file extension`. When `type` is set to `module`, Node.js expects the files to have specific extensions (e.g., `.cjs`, `.mjs`, or `.js`).

I was able to resolve this issue by setting the `"module": "CommonJS"` option in the `tsconfig.json` file and avoiding the use of `"type": "module"` in the `package.json` file. However, I believe setting `"module": "CommonJS"` might not be ideal, as it allows mixed usage of `import` and `require` within a single file.

--- 

### Using the esm loader